components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -64,20 +74,387 @@ pub struct SplitResult {
+            approximate_keys,
+            approximate_size,
+            locks,
+            last_split_region,
+        } = Box::into_inner(init_info);
+
+        let mut need_schedule_apply_fsm = false;
+        let region_id = region.id;
+
+        let replace = region.get_region_epoch().get_version()
+            > self
+                .storage()
+                .region_state()
+                .get_region()
+                .get_region_epoch()
+                .get_version();
+
+        if !self.storage().is_initialized() || replace {
+            let mut wb = store_ctx.engine.log_batch(5);
+            write_initial_states(&mut wb, region.clone()).unwrap_or_else(|e| {
+                panic!(
+                    "{:?} fails to save split region {:?}: {:?}",
+                    self.logger.list(),
+                    region,
+                    e
+                )
+            });
+
+            // todo: need to do asynchronously?
+            store_ctx.engine.consume(&mut wb, true).unwrap_or_else(|e| {
+                panic!(
+                    "{:?} fails to consume the write: {:?}",
+                    self.logger.list(),
+                    e,
+                )
+            });
+
+            let split_temp_path = store_ctx.tablet_factory.tablet_path_with_prefix(
+                SPLIT_PREFIX,
+                region_id,
+                RAFT_INIT_LOG_INDEX,
+            );
+            let tablet_path = store_ctx
+                .tablet_factory
+                .tablet_path(region_id, RAFT_INIT_LOG_INDEX);
+            std::fs::rename(&split_temp_path, &tablet_path).unwrap_or_else(|e| {
+                panic!(
+                    "{:?} fails to rename from tablet path {:?} to normal path {:?} :{:?}",
+                    self.logger.list(),
+                    split_temp_path,
+                    tablet_path,
+                    e
+                )
+            });
+
+            let storage = Storage::new(
+                region_id,
+                store_ctx.store_id,
+                store_ctx.engine.clone(),
+                store_ctx.read_scheduler.clone(),
+                &store_ctx.logger,
+            )
+            .unwrap_or_else(|e| panic!("fail to create storage: {:?}", e))
+            .unwrap();
+
+            let applied_index = storage.apply_state().get_applied_index();
+            let peer_id = storage.peer().get_id();
+            let raft_cfg = raft_config(peer_id, applied_index, &store_ctx.cfg);
+
+            let mut raft_group = RawNode::new(&raft_cfg, storage, &self.logger).unwrap();
+            // If this region has only one peer and I am the one, campaign directly.
+            if region.get_peers().len() == 1 {
+                raft_group.campaign().unwrap();
+                self.set_has_ready();
+            }
+            self.set_raft_group(raft_group);
+
+            need_schedule_apply_fsm = true;
+        } else {
+            // todo: when reaching here, it is much complexer.
```

Comment:
what's the logic to do here?

Reply:
If the peer is initialized and moreover, new than the information sent from the parent.